GVRf/Framework/backend_oculus/src/main/java/org/gearvrf/OvrViewManager.java

```diff
@@ -348,6 +352,10 @@ void onDrawEye(int eye) {
     protected void onDrawFrame() {
         beforeDrawEyes();
         drawEyes(mActivity.getActivityNative().getNative());
+
+        // update the gear controller
```

Comment:
Have you tried to measure the overhead in the case there is no controller? In my non-scientific test this takes ~0.06ms which is not that much but then it is not free either. Have seen it spike a lot too.

We can merge it like this but please examine it again in the future. You could call the controller's onDrawFrame from the native GearController::onFrame directly, if there is a controller. The check by findConnectedGearController is consistently fast.

Also the Java Buffers may not always be the universally best solution. I guess you need to have a large enough of payload to gain something out of them. For what we are passing around, copies may be simpler and faster.